services/comments.js

```diff
@@ -51,50 +51,76 @@ module.exports = class CommentsService {
 
   /**
    * Edit a Comment
-   * @param {CommentModel|String} comment you want to edit (or its ID)
-   * @param {String} body  the new Comment body
-   * @param {String} status  the new Comment status
+   * @param {String} id    comment.id you want to edit (or its ID)
+   * @param {String} asset_id   asset_id of the comment
+   * @param {String} editor     user.id of the user trying to edit the comment (will err if not comment author)
+   * @param {String} body       the new Comment body
+   * @param {String} status     the new Comment status
    */
-  static async edit(comment, {body, status, ignoreEditWindow}) {
-    if (typeof comment === 'string') {
-
-      // it's an id
-      comment = await this.findById(comment);
-    }
-    const editWindowExpired = (comment) => {
-      const now = new Date;
-      const editableUntil = this.getEditableUntilDate(comment);
-      return now > editableUntil;
-    };
-    if (( ! ignoreEditWindow) && editWindowExpired(comment)) {
-      throw Object.assign(new Error('Edit window is over.'), {
-        name: 'EditWindowExpired'
-      });
-    }
+  static async edit(id, asset_id, editor, {body, status, ignoreEditWindow}) {
     if (status && ! STATUSES.includes(status)) {
       throw new Error(`status ${status} is not supported`);
     }
-
-    const {id} = comment;
-    const {nModified} = await CommentModel.update({id}, {
-      $set: {
-        body,
-        status,
+    const lastEditableCommentCreatedAt = new Date((new Date()).getTime() - EDIT_WINDOW_MS);
+    const filter = Object.assign(
+      {
+        id,
+        asset_id,
+        author_id: editor,
       },
-      $push: {
-        body_history: {
+      ignoreEditWindow ? {} : {
+        created_at: {
+          $gt: lastEditableCommentCreatedAt,
+        },
+      }
+    );
+    const {nModified} = await CommentModel.update(
+      filter,
+      {
+        $set: {
           body,
-          created_at: new Date(),
+          status,
         },
-        status_history: {
-          type: status,
-          created_at: new Date(),
-        }
-      },
-    });
+        $push: {
+          body_history: {
+            body,
+            created_at: new Date(),
+          },
+          status_history: {
+            type: status,
+            created_at: new Date(),
+          }
+        },
+      }
+    );
     switch (nModified) {
-    case 0:
-      throw new Error(`Couldn't edit comment. There is no Comment with id "${id}"`);
+    case 0: {
+
+          // disambiguate possible error cases
+      const comment = await this.findById(id);
```

Comment:
> This error is already resolved by the frontend

This is only true insofar as coral-embed-stream is the only 'frontend' that wants to talk with the CommentsService. Right now that's mostly true. But if this project is successful, there will also be:

* Alternative web clients for customers who want different design/ux/etc
* Native mobile app clients
* Forks who want to add features using the services directly

Developing alternative clients (and even other talk features that want to use this service method) is significantly easier with useful error messages. 

The 'cost' of this call right now is a couple milliseconds in a very rare case.

The 'cost' of any other user of CommentService trying to figure out what to do with an ambiguous error is several orders of magnitude more. Let's say one of those use cases involves a requirement that the end-user can see a message explaining that their edit failed because the time window was up (as opposed to another reason). Then the 'cost' to that user also involves reintroducing this code anyway (and having this same discussion with you).

I've never read an API Design Guide that didn't include a principle like "Return useful error messages". I'm all for ensuring an optimized happy path, but removing this part is optimizing for theory and fractions of a penny of CPU time at the expense of real humans.